ui/page/dcrdex/market.go

```diff
@@ -1928,8 +1970,10 @@ func (pg *DEXMarketPage) orderWithLots() bool {
 	return !pg.switchLotsOrAmount.IsChecked()
 }
 
-func (pg *DEXMarketPage) noSupportedMarket() bool {
-	return pg.marketSelector.Selected() == values.String(values.StrNoSupportedMarket)
+// noMarket checks if the selected server is disconnected or has no supported
+// markets.
+func (pg *DEXMarketPage) noMarket() bool {
+	return pg.marketSelector.Selected() == values.String(values.StrNoSupportedMarket) || pg.marketSelector.Selected() == values.String(values.StrDEXServerDisconnected)
 }
```

Comment:
This is a weird check. Why checking the string instead of the dex client?

Reply:
This is what the user sees atm, and I think it's safe to base our judgment on that. If there xc has reconnected, the page is instantly updated.